WinAppDriver/CommandHandlers/NewSessionCommandHandler.cs

```diff
@@ -61,15 +63,57 @@ public override Response Execute(CommandEnvironment environment, Dictionary<stri
             var desiredCapabilities = JObject.Parse(parameters["desiredCapabilities"]?.ToString() ?? "{}").ToObject<Dictionary<string, object>>();
             //parameters["desiredCapabilities"] as Dictionary<string, string> ?? new Dictionary<string, string>();
 
-            if (!desiredCapabilities.TryGetValue("processName", out var processName))
+            if (!desiredCapabilities.TryGetValue("processName", out var processName) 
+                & !desiredCapabilities.TryGetValue("exePath", out var exePath))
             {
-                return Response.CreateMissingParametersResponse("processName");
+                return Response.CreateMissingParametersResponse("processName or exePath");
             }
 
-            var process = Process.GetProcessesByName(processName.ToString()).FirstOrDefault();
-            if (process == null)
+
+            Process process = null;
+            if (processName != null)
             {
-                return Response.CreateErrorResponse(-1, $"Cannot attach to process '{processName}', no such process found.");
+                
+                process = Process.GetProcessesByName(processName.ToString()).FirstOrDefault();
+
+                if (process == null)
+                {
+                    var processes = Process.GetProcesses();
+                    process = processes.Where(x =>
+                    {
+                        return x.ProcessName.Contains(processName.ToString())
+                        || processName.ToString().Contains(x.ProcessName);
+                    }).FirstOrDefault();
+                }
+
+                if (process == null && exePath == null)
+                {
+                    return Response.CreateErrorResponse(-1, $"Cannot attach to process '{processName}', no such process found.");
+                }
+            }
+
+            if (process == null && exePath != null)
```

Comment:
Please move this functionality into a new file/class (including the StartProcessFromPath method)

Reply:
Okay, I will do it. So do you decided to leave the process name as a required parameter or agree with my suggestion of choice between the path to executable and the process name?

Reply:
I agree with the option to either attach (`processName`) or launch the program (`execPath`) when not found (the question is how propagate the command line arguments).

Reply:
Because of exception handling in the outer method, I removed the try-catch statement.